WordPress/gutenberg
# DropdownMenu V2: add support for rendering in legacy popover slot

Part of #50459

## What?


This PR introduces changes that allow the the new experimental version of `DropdownMenu` to render in the same slots as the legacy `Popover`-based components (including current stable `DropdownMenu`)

## Why?


For better backwards compatibility between the new and the legacy version of the component

## How?


- The slot name can be passed either via a `slotName` prop, or via the popover slot name context
- When a slot name isn't specified, the default popover slot name is used
- If a slot associated to the computed slot name is found, the menu's popover is rendered in that slot. Otherwise the component behaves as the ariakit default.

## Testing Instructions





- Download the changes from this PR and run storybook locally
- Visit the temporary Storybook examples that have been added to the new experimental `DropdownMenu` component. Open the dropdown menu and make sure that the menu is rendered correctly in the popover slot component (you can do so by inspecting the DOM and comparing it against the default story)
- Compare the results from the point above to the equivalent Storybook examples for the legacy `DropdownMenu` component. The two components should behave the same way when picking the solt in which to render their popover.
